src/main/kotlin/com/velocitypowered/api/kt/proxy/connection/ConnectionRequestExt.kt

```diff
@@ -0,0 +1,9 @@
+package com.velocitypowered.api.kt.proxy.connection
+
+import com.velocitypowered.api.proxy.connection.Player
+import com.velocitypowered.api.proxy.player.ConnectionRequestBuilder
+import com.velocitypowered.api.proxy.server.RegisteredServer
+import kotlinx.coroutines.future.await
+
+suspend fun Player.connect(server: RegisteredServer): ConnectionRequestBuilder.Result =
+  createConnectionRequest(server).connect().await()
```

Comment:
This could also be a method in velocity api, and just return `CompletableFuture`. `await` can be called on CompletableFuture to make it suspending.

Reply:
await in CompletableFuture is blocking method. This violates the principles of kotlin-coroutine, which is why you need to wrap it in suspendCoroutine